logstash-core/lib/logstash/plugin.rb

```diff
@@ -8,8 +8,26 @@
 require "logstash/plugins/registry"
 
 class LogStash::Plugin
+
+
+  class Timer
+
+    def initialize
+      @timer = 0
+    end
+
+    def start
```

Comment:
Time is pretty hard to test, so I wonder if we can pass a time as an argument:

```ruby
def start(time=Time.now)
  @timer = time
end
```

This will have the same behavior as the current method, and adds the benefit that we can fake time more easily in testing.